Source/Core/Core/Boot/Boot.cpp

```diff
@@ -271,199 +292,188 @@ bool CBoot::Load_BS2(const std::string& boot_rom_filename)
   return true;
 }
 
-// Third boot step after BootManager and Core. See Call schedule in BootManager.cpp
-bool CBoot::BootUp()
+static const DiscIO::IVolume* SetDefaultDisc()
 {
-  SConfig& _StartupPara = SConfig::GetInstance();
+  auto& config = SConfig::GetInstance();
+  // load default image or create virtual drive from directory
+  if (!config.m_strDVDRoot.empty())
+    return SetDisc(DiscIO::CreateVolumeFromDirectory(config.m_strDVDRoot, config.bWii));
+  if (!config.m_strDefaultISO.empty())
+    return SetDisc(DiscIO::CreateVolumeFromFilename(config.m_strDefaultISO));
+  return nullptr;
+}
 
-  if (_StartupPara.m_BootType == SConfig::BOOT_BS2)
-    NOTICE_LOG(BOOT, "Booting %s", _StartupPara.m_strBootROM.c_str());
-  else
-    NOTICE_LOG(BOOT, "Booting %s", _StartupPara.m_strFilename.c_str());
+// Third boot step after BootManager and Core. See Call schedule in BootManager.cpp
+bool CBoot::BootUp(const BootParameters& boot)
+{
+  SConfig& config = SConfig::GetInstance();
 
   g_symbolDB.Clear();
 
   // PAL Wii uses NTSC framerate and linecount in 60Hz modes
-  VideoInterface::Preset(DiscIO::IsNTSC(_StartupPara.m_region) ||
-                         (_StartupPara.bWii && _StartupPara.bPAL60));
+  VideoInterface::Preset(DiscIO::IsNTSC(config.m_region) || (config.bWii && config.bPAL60));
 
-  switch (_StartupPara.m_BootType)
+  struct BootTitle
```

Comment:
Would this look better with a lambda and something like the [overload proposal](https://github.com/viboes/tags/blob/master/doc/proposals/overload/P0051R2.md#motivation)? We could put that into a header somewhere (the [implementation](https://gist.github.com/leoetlino/2272a70232dc14176403bd291f33a4fb) is quite short).